File: Gauntlet of the Hero - GOTH/Assets/_Scripts/Models/Backend/BattleSessionCreateRequestDto.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace GauntletOfTheHero.Backend.Models
{
    [System.Serializable]
    public class BattleSessionCreateRequestDto
    {
        [JsonProperty("heroId")]
        public long heroId;

        [JsonProperty("monsterId")]
        public long monsterId;

        [JsonProperty("heroMaxHealth")]
        public int? heroMaxHealth;

        [JsonProperty("heroCurrentHealth")]
        public int? heroCurrentHealth;

        [JsonProperty("heroAttack")]
        public int? heroAttack;

        [JsonProperty("heroDefense")]
        public int? heroDefense;

        [JsonProperty("heroMagic")]
        public int? heroMagic;

        [JsonProperty("heroMoveIds")]
        public List<long> heroMoveIds = new List<long>();
    }
}

File: nordeus_challenge_backend_spring_boot/src/main/java/com/mmileticc/nordeuschallenge/model/dto/BattleSessionCreateRequestDTO.java
package com.mmileticc.nordeuschallenge.model.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BattleSessionCreateRequestDTO {

    @NotNull(message = "heroId is required")
    @Positive(message = "heroId must be positive")
    private Long heroId;

    @NotNull(message = "monsterId is required")
    @Positive(message = "monsterId must be positive")
    private Long monsterId;

    private Integer heroMaxHealth;
    private Integer heroCurrentHealth;
    private Integer heroAttack;
    private Integer heroDefense;
    private Integer heroMagic;
    private List<Long> heroMoveIds = new ArrayList<>();
}
